editor/gui/scene_tree_editor.cpp

```diff
@@ -925,6 +925,27 @@ void SceneTreeEditor::_notification(int p_what) {
 
 			_update_tree();
 		} break;
+
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (is_visible()) {
+				TreeItem *item = nullptr;
+				if (selected) {
+					// Scroll to selected node.
+					item = _find(tree->get_root(), selected->get_path());
+				} else if (marked.size() == 1) {
+					// Scroll to a single marked node.
+					Node *marked_node = *marked.begin();
+					if (marked_node) {
+						item = _find(tree->get_root(), marked_node->get_path());
+					}
+				}
+
+				if (item) {
+					// Must wait until tree is properly sized before scrolling.
+					callable_mp(tree, &Tree::scroll_to_item).call_deferred(item, true);
```

Comment:
This PR seems to be causing crashes. Probably deferred call should use `ObjectID` instead of raw pointer, otherwise the tree item might be removed before callback execution and pointer will become invalid.

Reply:
Can you show me other code locations that do something like this? Also if there's a reliable way to reproduce the crash would help.

Reply:
> Can you show me other code locations that do something like this?

Not sure if there's exactly same situations, buy you can do something like:

Add a new callback:
```c++
void Tree::_scroll_to_item_id(ObjectID p_item_id, bool p_center_on_item) {
	TreeItem *item = Object::cast_to<TreeItem *>(ObjectDB::get_instance(p_item_id));
	if (item) {
		scroll_to_item(item, p_center_on_item);
	}
}
```

And call it like this
```c++
if (item) {
	ObjectID item_id = item->get_instance_id();
	callable_mp(tree, &Tree::_scroll_to_item_id).call_deferred(item_id, true);
}
```